packages/amplify-graphql-api-construct/src/__tests__/__functional__/combine-synthesis.test.ts

```diff
@@ -0,0 +1,602 @@
+    `;
+    const sqlStrategy1 = mockSqlDataSourceStrategy({ name: 'sqlDefinition1' });
+    const sqlStrategy2 = mockSqlDataSourceStrategy({ name: 'sqlDefinition2' });
+    const definition1 = AmplifyGraphqlDefinition.fromString(postSchemaSql, sqlStrategy1);
+    const definition2 = AmplifyGraphqlDefinition.fromString(tagSchemaSql, sqlStrategy2);
+    expect(() => makeApiByCombining(definition1, definition2)).toThrow();
+  });
+
+  // This is technically redundant with tests in the relational transformer (the test only uses a single data source, so it's not really a
+  // `combine` test), but it makes sense to add it here since we're testing other combinations
+  it('fails if a many-to-many relationship is declared in a single SQL data source', () => {
+    const schema = /* GraphQL */ `
+      type Post @model {
+        id: ID! @primaryKey
+        title: String!
+        content: String
+        tags: [Tag] @manyToMany(relationName: "PostTags")
+      }
+      type Tag @model {
+        id: ID! @primaryKey
+        label: String!
+        posts: [Post] @manyToMany(relationName: "PostTags")
+      }
+    `;
+    const sqlStrategy1 = mockSqlDataSourceStrategy({ name: 'sqlDefinition1' });
+    const definition1 = AmplifyGraphqlDefinition.fromString(schema, sqlStrategy1);
+    expect(() => makeApiByCombining(definition1)).toThrow();
+  });
+
+  it('supports definitions with both models and custom SQL statements', () => {
+    const sqlStrategy1 = mockSqlDataSourceStrategy({ name: 'sqlstrategy1' });
+    const sqlStrategy2 = mockSqlDataSourceStrategy({
+      name: 'sqlstrategy2',
+      dbType: 'POSTGRES',
+      customSqlStatements: {
+        customSqlQueryReference: 'SELECT 1',
+      },
+    });
+    const definition1 = AmplifyGraphqlDefinition.fromString(SCHEMAS.todo.sql, sqlStrategy1);
+    const definition2 = AmplifyGraphqlDefinition.fromString(SCHEMAS.customSqlQueryReference, sqlStrategy2);
+
+    const api = makeApiByCombining(definition1, definition2);
+
+    const {
+      resources: {
+        cfnResources: { cfnGraphqlApi, cfnGraphqlSchema, cfnApiKey, cfnDataSources },
+        functions,
+      },
+    } = api;
+
+    expect(cfnGraphqlApi).toBeDefined();
+    expect(cfnGraphqlSchema).toBeDefined();
+    expect(cfnApiKey).toBeDefined();
+    expect(cfnDataSources).toBeDefined();
+
+    const ddbDataSources = Object.values(cfnDataSources).filter((dataSource) => dataSource.type === 'AMAZON_DYNAMODB');
+    expect(ddbDataSources.length).toEqual(0);
+
+    const lambdaDataSources = Object.values(cfnDataSources).filter((dataSource) => dataSource.type === 'AWS_LAMBDA');
+    expect(lambdaDataSources.length).toEqual(2);
+
+    // Expect one SQL Lambda function per strategy
+    expect(functions).toBeDefined();
+    expect(functions['SQLFunctionsqlstrategy1']).toBeDefined();
+    expect(functions['SQLFunctionsqlstrategy2']).toBeDefined();
+  });
+
+  it.skip('supports Query definitions split amongst heterogeneous definitions', () => {
```

Comment:
This is the `There can be only one type named "Query"` error?

Reply:
Correct. Am in process of fixing that now